test/e2e/util.go

```diff
@@ -541,6 +552,33 @@ func waitForPodCondition(c *client.Client, ns, podName, desc string, timeout tim
 	return fmt.Errorf("gave up waiting for pod '%s' to be '%s' after %v", podName, desc, timeout)
 }
 
+// waitForMatchPodsCondition finds match pods based on the input ListOptions.
+// waits and checks if all match pods are in the given podCondition
+func waitForMatchPodsCondition(c *client.Client, opts api.ListOptions, desc string, timeout time.Duration, condition podCondition) error {
```

Comment:
Can you just reuse the pattern/functions here: https://github.com/kubernetes/kubernetes/blob/master/test/e2e/util.go#L2258?